### DISCUSSION TITLE
When update state only last vue component is rerendered

### DISCUSSION THREAD

**Discussion by fdimone** [alovajs/alova#357] [Q&A] [answered]
Hi guys,

In the App.vue i render two instance of the same component, which displays a random numbers list fetched from an API (with same URL params), when i update state (using updateState) or refetch the API used in the component list (using useFetcher) i expect to update both components with the new state but only the last component rendered get updated... is this a normal behaviour, my mistake or it can be an internal bug?

After first fetch:
![image](https://github.com/alovajs/alova/assets/114220775/0ee6ba3a-83b0-45e9-b80b-2040819eccca)

After refetching: 
![image](https://github.com/alovajs/alova/assets/114220775/e411b85f-b8f5-4e64-acb9-bae41959314c)

[Reproduction ](https://stackblitz.com/edit/vitejs-vite-mkfhsr?file=src%2FApp.vue)(using alova 2.20.3).

Thanks

**Comment by MeetinaXD** [comment]
Maybe I know why...

This is because alova stores only one hook state globally for the same key.

Detailed description:
1. When using hooks (useRequest/useWatcher/useFetcher), alova creates state object for them.
2. alova gets a key from the method and places the state object in StateCache based on the key
3. When you use useFetcher requests, it retrieves the state object from StateCache and puts the latest data into `state.data`
4. Since the key is the same, the state cache created with useRequest() will always be overwritten by the next one

The relevant code is here: [useHookToSendRequest.ts#L138](https://github.com/alovajs/alova/blob/65c64b86616b65485a6bd12e42a3da56683d5dea/src/functions/useHookToSendRequest.ts#L138)

I'm not sure if this is a feature or a bug. @JOU-amjs

**Reply by fdimone** [reply]
Thanks @MeetinaXD.

I ended up to use scene-vue actionDelegationMiddleware and accessAction methods to force refetch for all states.

By my side i think that is strange behaviour and the solution i found is not scalable at all.

This "problem" make difficult to use alova as a store management for the data fetched from the server(like in tanstack/vue-query)

**Reply by dux-web** [reply]
I'm also experiencing this issue, which makes it extremely inconvenient to update the state, which logically should update the same named data in the loaded component

**Comment by JOU-amjs** [comment, marked as answer]
now this bug has been fixed in 3.2.12
